From adcf99610bf084fd3cd3e3012e1d1be548de46b0 Mon Sep 17 00:00:00 2001
From: Pascal Quantin <pascal.quantin@gmail.com>
Date: Sat, 24 Oct 2015 11:32:16 +0200
Subject: [PATCH 66/90] SDP: protect against out of bounds access

Change-Id: I4b24441cb26757b639e8113cab18d64c7f07112f
Ping-Bug: 9887
Reviewed-on: https://code.wireshark.org/review/11241
Reviewed-by: Pascal Quantin <pascal.quantin@gmail.com>
(cherry picked from commit 2ddd92b6f8f587325b9e14598658626f3a007c5c)
Reviewed-on: https://code.wireshark.org/review/11261
(cherry picked from commit e1979529b682a3b9b0bcefe3a54058c5ae314c5b)
Reviewed-on: https://code.wireshark.org/review/13753
Reviewed-by: Balint Reczey <balint@balintreczey.hu>
---
 epan/dissectors/packet-sdp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/epan/dissectors/packet-sdp.c b/epan/dissectors/packet-sdp.c
index 7b5fad4..95c29b4 100644
--- a/epan/dissectors/packet-sdp.c
+++ b/epan/dissectors/packet-sdp.c
@@ -2600,7 +2600,7 @@ dissect_sdp(tvbuff_t *tvb, packet_info *pinfo, proto_tree *tree)
     /* Free the remainded hash tables not used */
     /* XXX - more placeholder functionality */
     if (transport_info == &local_transport_info) {
-        for (n = transport_info->media_count; n < SDP_MAX_RTP_CHANNELS; n++)
+        for (n = MAX(transport_info->media_count, 0); n < SDP_MAX_RTP_CHANNELS; n++)
         {
             if (!transport_info->media[n].set_rtp)
             {
-- 
2.1.4

